### DISCUSSION TITLE
How to Make a Column (the ID field) as Always the Last Order By Field

### DISCUSSION THREAD

**Discussion by benlwong** [yajra/laravel-datatables#2965] [Q&A] [answered]
I just learned that if the data has the same values for a column (e.g. a date field), the rows can be returned in any order, from one sort to the next. So, this inconsistency has caused some rows to appear from one page to the next (via server-side paging). To fix this issue, I need to provide a field where the ordering does not change, and where each row is uniquely identified. 

So, my question is how do I make a specific field (e.g. the ID column) as always the last column to be sorted? For example, if the user selects the date column to be sorted, then the query will sort by date, then by ID.

In my DataTable code, I have the following method, but I am not sure if this is the place (or how) to make this happen:

```
protected function getBuilderParameters()
{
   return [
         'order' => [[0, 'asc']],
         'dom' => "<'row'<'col-md-12'B>>
                   <'row'<'col-md-6'l><'col-md-6'f>>
                   <'row'<'col-md-12'tr>>
                   <'row'<'col-md-6'i><'col-md-6'p>>",
         'buttons' => ['csv', 'excel', 'print', 'reload'],
   ];
}
```

Any pointers or advice would be greatly appreciated.

**Comment by yajra** [comment, marked as answer]
I think what you need is [orderFixed](https://datatables.net/reference/option/orderFixed)

**Reply by benlwong** [reply]
Thanks!
